test-schema.json

```diff
@@ -27,6 +27,16 @@
                 "type": "array",
                 "items": { "$ref": "#/$defs/test" },
                 "minItems": 1
+            },
+            "specification":{
+                "description": "Location of the test case in the specification",
+                "type": "object",
+                "minProperties": 1,
+                "properties":{
+                    "jsonschema": {"type": "string"},
+                    "rfc": {"type": "string"}
```

Comment:
Are these both URIs?

Reply:
They are URL's for a particular section in the JSON Schema Spec. They will redirect to the location where test-case is more relevant

Reply:
Is the structure of "specification" property supposed to be something else

Reply:
1. These are supposed to be informative comments.  It's not really anything that needs to be linked to.  We really just need some sort of reference.  I think just the spec name and section number will suffice.
    ```
    "additionalProperties":
    {
      "type": "string",
      "pattern": "^[0-9](\.[0-9])*$"}
    }
    ```
    which would give something like
    ```
    {
      "core": "8.5.2",
      "validation": "6.3"
    }
    ```
2. What are "jsonschema" and "rfc" intended to refer to?

Reply:
2.. Actually each section of the draft has many paragraphs, and each paragraph has unique URI's. 
like, https://json-schema.org/draft/2020-12/draft-bhutton-json-schema-01#section-10.3.2.3-2 URI will redirect to paragraph 2 of "additionalProperties" section. 
So, I thought why not that can be made use. And rfc -> too an URI for any section of RFC the description of the test located in.

1. Wouldn't it make test cases bigger. And also we can provide more precise location using URLs. So, why not URI's?
 - as we may need to provide multiple uris?
 - usability of 
 ```{
  "core": "8.5.2",
  "validation": "6.3"
}``` this will be greater??

Reply:
`https://json-schema.org/draft/2020-12/draft-bhutton-json-schema-01#section-10.3.2.3-2`

If you look at this format it has a common pattern.

- `https://json-schema.org/draft/2020-12/draft-bhutton-json-schema-01#section-`
- `<section>-<para>`

I think it'd be better just to include the last part of that, the section and paragraph.  If desired, tooling can construct the URL.

Again, this is just a reference for documentation purposes. It doesn't need to be functional.

Reply:
I totally got that. We can have spec name: core/validation and for rfc: rfc3456/rfc4567 or ...
Then we can have---> section
                             ---> para
So, mostly 3-4 indicators. Right? ❤️‍🔥

Reply:
@Julian what is the philosophy of tests for requirements like "behaves per _X_ spec, section _Y_"?  Do we include tests checking behavior against other specs?

Reply:
Like why would we include that info? Yeah we definitely have lots of that, see e.g. all the format tests, all of which essentially are testing behavior specified in some other specification and generally RFC.

But def fine to leave it out to start I think.

(I haven't read the rest of the thread yet!)

Reply:
Can the structure be:
```
{
    "specification":{
        "jsonschema-validation":{
              "section": "6.3.1",
              "para": "3"
        },
        "rfc8259":{
            "section": 7
        },
        // "jsonschema-core":{...}
    }
} 
```
and because there may be more than one place in spec, we can make for example:
```
        "jsonschema-validation":[{
              "section": "6.3.1",
              "para": "3"
        }, 
        {
            "section": "6.3.1",
             "para": "4"
        }]
```
can keep it such that  `"jsonschema-validation"` can take both array and object.. 

And instead of asking can I make as many commits as possible and then take comments?